src/node-software/configuration.md

```diff
@@ -0,0 +1,59 @@
+# Configuration
+
+Node software needs configuration. We generally run multiple nodes, with similar
+but slightly different configuration.
+
+#### Use a configuration file. {.p3 #config-file}
+We don’t have a very strong opinion on how to configure software.
+Command-line flags, [environment variables][12factor],
+or a configuration file are all fine.
+However, we found that as software matures and grows more complex,
+it needs more configuration options,
+and putting everything on the command line can become unwieldy.
+Furthermore,
+in command-line arguments and environment variables,
+everything is a string,
+while configuration files enable more structured data.
+We recommend using [TOML][toml] over YAML,
+as TOML contains fewer footguns and implementation ambiguities.
+
+[12factor]: https://12factor.net/config
```

Comment:
oohh, i learned to hate the 12factor overtime, but it's whatever

hope people won't think we advertise it in any way